Source/Core/Core/NetPlayServer.h

```diff
@@ -16,6 +17,7 @@
 #include "Common/Timer.h"
 #include "Common/TraversalClient.h"
 #include "Core/NetPlayProto.h"
+#include "Core/NetPlayQoS.h"
```

Comment:
This can be a forward declaration as far as I can tell.

Reply:
Not when using a `std::unique_ptr`.

Reply:
If you default Client's destructor in the cpp file as

```cpp
NetPlayServer::Client::~Client() = default;
```

it will likely compile fine. The reason this doesn't work as is, is because non-explicit destructors are inline by default. In this case, it will cause `Client`'s destructor (which needs to check whether or not to call the `std::unique_ptr`'s deleter) to be generated where the compiler can only see the incomplete type.